### PR TITLE
Changes to use vector rather than unordered_map for OSM nodes

### PR DESCRIPTION
**Pull request by dnesbitt61**
Allows all of North America to be loaded into osmdata and still be below 10GB. 
Also hardcoded to 1 thread for now - threading seems to incur swap more readily.
The node tests are commented out...not sure what the behavior will be if the node is not found.

### CODE REVIEW

File: src/mjolnir/graphbuilder.cc [hunk 1]
```diff
@@ -1107,7 +1121,7 @@ std::vector<SignInfo> GraphBuilder::CreateExitSignInfoList(
 // Build tiles for the local graph hierarchy
 void GraphBuilder::BuildLocalTiles(const uint8_t level, const OSMData& osmdata) const {
   // A place to hold worker threads and their results, be they exceptions or otherwise
-  std::vector<std::shared_ptr<std::thread> > threads(threads_);
+  std::vector<std::shared_ptr<std::thread> > threads(1); //threads_);
```
Comment (kevinkreiser) [inline src/mjolnir/graphbuilder.cc]: i think you meant to change this back. if you want to run mjolnir with only 1 thread set it that way in the config

**Comment by gknisely** [comment]
:ship: after u fix what kevin noted for the config.

File: valhalla/mjolnir/osmdata.h [hunk 2]
```diff
@@ -64,6 +66,17 @@ struct OSMData {
   // Names
   UniqueNames name_offset_map;
 
+  OSMNode GetNode(const uint64_t osmid) {
```
Comment (kevinkreiser) [inline valhalla/mjolnir/osmdata.h]: should make this method const

File: src/mjolnir/graphbuilder.cc [hunk 3]
```diff
@@ -138,7 +140,7 @@ Node& GraphBuilder::GetNode(const GraphId& graphid) {
 }
 
 // Construct edges in the graph and assign nodes to tiles.
-void GraphBuilder::ConstructEdges(const OSMData& osmdata, const float tilesize) {
```
Comment (kevinkreiser) [inline src/mjolnir/graphbuilder.cc]: should pass this by const ref after making GetNode const

File: valhalla/mjolnir/osmdata.h [hunk 4]
```diff
@@ -64,6 +66,17 @@ struct OSMData {
...
+    OSMNode test(osmid, 0.0f, 0.0f);
+    auto it = std::equal_range(nodes.begin(), nodes.end(), test);
+    return *(it.first);
+  }
+
+  // Sort the OSM nodes
+  void SortNodes() {
```
Comment (kevinkreiser) [inline valhalla/mjolnir/osmdata.h]: i think we should remove this method, and then in pbfgraphparser.load, do it directly there. that way anyone who gets an osmdata object won't have to know to call this method.

Comment (dnesbitt61) [inline valhalla/mjolnir/osmdata.h]: yea, was thinking that too...

**Pull request merged**